torchci/clickhouse_queries/recent_pr_workflows_query/query.sql

```diff
@@ -1,84 +1,83 @@
--- !!! Query is not converted to CH syntax yet.  Delete this line when it gets converted
 -- This workflow is used by Dr.CI to get all the jobs from pull requests. The failures will then be
 -- classified into new failures and unrelated failures such as broken trunk, flaky, unstable, etc.
 WITH relevant_shas as (
-  select j.head_sha
-  from workflow_job j
-  where
-    PARSE_TIMESTAMP_ISO8601(j.completed_at) > (
-      CURRENT_TIMESTAMP() - MINUTES(: numMinutes)
-    )
-    AND :prNumber = 0
-  union
-  select pr.head.sha as head_sha
-  from commons.pull_request pr
-  where pr.number = :prNumber
+  select head_sha
+  from materialized_views.workflow_job_by_completed_at
```

Comment:
I think we should version tracking the query to create those materialized view as it's not clear how they are created.  What do you think?

Reply:
I would like to, but I don't know a good way to replace table definitions without needing to remake the entire table, so I'm not sure how useful version tracking would be if we can't enforce it

Reply:
I see.  Let's chat more about this topic next week.  In my past experience working with relational databases, the way to enforce this is usually to:

1.  Revoke the generic permission to create new table in the default database while still allows folks to experiment in their own database
2. Maintain the SQL query script to recreate the table and view, usually in the following format
```
IF TABLE NOT EXIST, CREATE TABLE ...;

ALTER TABLE ...;
ALTER TABLE ...;
```
3. Have a workflow with sufficient permission to run the script from step 2

This will be a great BE item.  IMO, this is similar to manage all AWS resources using Terraform instead manually creating them like what we used to do.